api/secret.go

```diff
@@ -302,7 +302,15 @@ func ParseSecret(r io.Reader) (*Secret, error) {
 	// First read the data into a buffer. Not super efficient but we want to
 	// know if we actually have a body or not.
 	var buf bytes.Buffer
-	_, err := buf.ReadFrom(r)
+
+	// io.Reader is treated like a stream and cannot be read
+	// multiple times. Duplicating this stream using TeeReader
+	// to use this data in case there is no top-level data from
+	// api response
+	var teebuf bytes.Buffer
+	tee := io.TeeReader(r, &teebuf)
```

Comment:
Do we have to read twice? It looks like we could possibly just make a copy of the original Decode response, rather than calling again. Or would that not work for some reason?

Reply:
I tried it, once the buffer is used to decode, the pointer reaches the end of the buffer. If we try to decode this again, the data in the buffer is nil.